scripts/zones/Port_Jeuno/npcs/Squintrox_Dryeyes.lua

```diff
@@ -3,57 +3,348 @@
-    local count         = trade:getItemCount()
-    local sLux          = trade:hasItemQty(xi.item.SEEDSPALL_LUX, 1)
-    local sLuna         = trade:hasItemQty(xi.item.SEEDSPALL_LUNA, 1)
-    local sAstrum       = trade:hasItemQty(xi.item.SEEDSPALL_ASTRUM, 1)
-    local acpMission    = player:getCurrentMission(xi.mission.log_id.ACP)
-    local hasCrimsonKey = player:hasKeyItem(xi.ki.CRIMSON_KEY)
-    local lastCrimson   = player:getCharVar('LastCrimsonKey') -- When last Crimson key was obtained
+-- maps the menu options to the key items/missions that block access
+-- second page is auto indexed as a list and matches the order of the 2nd page menu
+-- first page is keyed by the key item, since the menu options don't have a clean pattern
+-- first page mostly listed here to have one place to reference all the metadata about a menu option's flow, but also to clean up some of the complex event args
+local menuMetadata =
+{
+    [1] =
+    {
+        ['initialList'] =
+        {
+            xi.ki.CRIMSON_KEY,
+            xi.ki.VIRIDIAN_KEY,
+            xi.ki.WHITE_CORAL_KEY,
+            xi.ki.BLUE_CORAL_KEY,
+            xi.ki.BLACK_CORAL_KEY,
+            xi.ki.MOOGLE_KEY,
+            xi.ki.BIRD_KEY,
+            xi.ki.BOMB_KEY,
+        },
+        [xi.ki.CRIMSON_KEY] =
+        {
+            expansion    = 1,
+            charVar      = 'LastCrimsonKey',
+            reqItems =
+            {
+                xi.item.SEEDSPALL_LUX,
+                xi.item.SEEDSPALL_LUNA,
+                xi.item.SEEDSPALL_ASTRUM,
+            },
+        },
+        [xi.ki.VIRIDIAN_KEY] =
+        {
+            expansion = 1,
+            charVar   = 'LastViridianKey',
+            reqKeyItems =
+            {
+                xi.ki.BOWL_OF_BLAND_GOBLIN_SALAD,
+                xi.ki.JUG_OF_GREASY_GOBLIN_JUICE,
+                xi.ki.CHUNK_OF_SMOKED_GOBLIN_GRUB,
+            }
+        },
+        [xi.ki.WHITE_CORAL_KEY] =
+        {
+            expansion = 2,
+            charVar   = 'LastWhiteCoralKey',
+            reqItems =
+            {
+                xi.item.ORCISH_PLATE_ARMOR,
+                xi.item.QUADAV_BACKSCALE,
+                xi.item.YAGUDO_CAULK,
+            },
+        },
+        [xi.ki.BLUE_CORAL_KEY] =
+        {
+            expansion = 2,
+            charVar   = 'LastBlueCoralKey',
+            reqKeyItems =
+            {
+                xi.ki.STURDY_METAL_STRIP,
+                xi.ki.PIECE_OF_RUGGED_TREE_BARK,
+                xi.ki.SAVORY_LAMB_ROAST,
+            },
+        },
+        [xi.ki.BLACK_CORAL_KEY] =
+        {
+            expansion = 2,
+            charVar   = 'LastBlackCoralKey',
+            reqKeyItems =
+            {
+                xi.ki.MOLDY_WORMEATEN_CHEST,
+            },
+        },
+        [xi.ki.MOOGLE_KEY] =
+        {
+            expansion      = 3,
+            charVar        = 'LastMoogleKey',
+            reqItemCharVar = 'ASA_kit',
+            reqItems =
```

Comment:
This table is nice, but the expansion, charVar and reqItemCharVar keys aren't correctly aligned like the rest of the table - choose one style and stick to it